historical_data_api_client/get_historical_data.py

```diff
@@ -0,0 +1,49 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import json
+
+import requests
+
+
+def get_historical_device_profiles(domain_name: str, auth_headers: dict,
```

Comment:
Nit: when defining types, using `typing.Dict` is recommended over the bare `dict` class, because it also allows to specify the type of keys/values you want to use. For example:
```python
get_historical_device_profiles(domain_name: str, auth_headers: Dict[str, str], ...)
```